ludwig/utils/image_utils.py

```diff
@@ -163,6 +163,71 @@ def read_image_from_str(img: str, num_channels: Optional[int] = None) -> torch.T
         return None
 
 
+def get_image_read_mode_from_num_channels(num_channels: int) -> ImageReadMode:
+    """Returns the torchvision.io.ImageReadMode corresponding to the number of channels.
+
+    If num_channels is not recognized, returns ImageReadMode.UNCHANGED.
+    """
+    mode = ImageReadMode.UNCHANGED
+    if num_channels == 1:
+        mode = ImageReadMode.GRAY
+    elif num_channels == 2:
+        mode = ImageReadMode.GRAY_ALPHA
+    elif num_channels == 3:
+        mode = ImageReadMode.RGB
+    elif num_channels == 4:
+        mode = ImageReadMode.RGB_ALPHA
+    return mode
+
+
+def read_image_if_path(path: Any, num_channels: Optional[int] = None) -> Union[Any, torch.Tensor]:
+    """Gets an image if `path` is a path (e.g. a string).
+
+    If it is not a path, return as-is.
```

Comment:
What are the possible data types of path (if not a string)?

Reply:
This function is intended to be usable by `map`-like workflows, so `path` could be any value stored in a column. For image columns, I believe we officially support image features being passed in as either path (str) or torch.Tensor objects (pre-loaded images). So, in this case, `path` could be a `torch.Tensor`. In that case, we would want this function to be a no-op.

Reply:
I'd recommend adding your response:

> ...path could be a torch.Tensor. In that case, we would want this function to be a no-op.

 as a comment, to make the intent clear here.

Reply:
How about the name the parameter `path_or_image` (instead of just `path`)